### PR TITLE
Disable precompiled headers for Linux/macOS by default

### PR DESCRIPTION
**Pull request by daschuer**
It turns out that we suffer longer build times during development, because touching a header forces a full rebuild.
See also #12081 
This makes also sure that CMAKE_DISABLE_PRECOMPILE_HEADERS is part of CMakeCache.txt in any case.

### CODE REVIEW

**Review by JoergAtGithub — CHANGES_REQUESTED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -519,6 +519,15 @@ if(WIN32)
   set(CMAKE_FIND_LIBRARY_PREFIXES "" "lib")
 endif()
 
+if(NOT DEFINED CMAKE_DISABLE_PRECOMPILE_HEADERS)
+  if(CMAKE_BUILD_TYPE STREQUAL "Debug")
```
[the remark this answers is not in the document]
Comment (ronso0) [reply CMakeLists.txt:523]: Then let's just turn it on by default, but at least it's now in Cmakelist and can be read e.g. by CmakeGUI.

Either way, a note in the Compiling wiki pages would be very helpful for contributors I think.

**Comment by JoergAtGithub** [comment]
> Why has pch such a big impact on your system? 

Well, that's what it supposed to do - reducing the build time. I think the better question is, why it doesn't reduce the build time on your system.


> What is the issue that makes #12371 taking more time?

I guess the precompiled header becomes rather large, because you included everything into one header.


> What happens if you change between branches that are touching a header that is part of the pch file?

What you can expect from a build system - rebuilding the files that differ and their dependencies. Same as without pch, just faster.

**Comment by daschuer** [comment]
Here is my benchmark, the Linux values are form my local builds, the Windows values from the CI. The absolute values are not very reliable, because I have not done repetitions. However the magnitude is already significant:   
|                           | Windows  | Linux    | Windows | Linux |
| ------------------------- | -------- | -------- | ------- | ----- |
| No ccache; No PCH;, clean | 25:41:00 | 53:36:00 | 100 %   | 100 % |
| ccache, No PCH, clean     | 33:24:00 | 60:12:00 | 130 %   | 112 % |
| no ccache; PCH;, clean    | 06:30:00 | 39:26:00 | 25 %    | 74 %  |
| ccache; PCH;, clean       | 06:18:00 | 49:14:00 | 25 %    | 92 %  |
| hot ccache:No PCH clean   | 06:33:00 | 03:48:00 | 26 %    | 7 %   |

Takeaways: 
* The ccache on Linux works more than double as fast as the sccache solution on Windows
* PCH has a more than double effect on Windows than on Linux 
* With PCH enabled a clean build is faster than a fully cached build. 

Conclusion: 
* Let's always use PCH on windows and disable sccache 
* Let's enable PCH on Linux only if no ccache is installed.

**Comment by JoergAtGithub** [comment]
Could you you disable sccache in the GitHub build.yml for Windows too. In CI, the preparation tasks take between [30s ](https://github.com/mixxxdj/mixxx/actions/runs/7255019241/job/19764848360 )and [4 minutes](https://github.com/JoergAtGithub/mixxx/actions/runs/7252748967/job/19757862930) CI time on top of the build time you measured.

**Comment by JoergAtGithub** [comment]
Did you also benchmark macOS? The build times of the last CI run look extreme: ARM 3 minutes, Intel 30 minutes
Linux cache restore failed with `Cache not found for input keys: ubuntu-22.04-ccache--3201, ubuntu-22.04-ccache-, ubuntu-22.04-ccache`. Wasn't it supposed to enable PCH if CCACHE is not available?
Windows build setup is fine now!

**Comment by daschuer** [comment]
For some reasons there is no Cache for Ubuntu. Maybe we have an issue with the two Ubuntu runs? 
The cache is enabled if ccache is installed. It is unrelated to the question if the workflow has a cache stored. It should create one if it is empty. The macOS Intel version has a view less cache hits. But I have no Idea if that explains the 40 m

**Comment by JoergAtGithub** [comment]
macOS x64 with PCH had ~18minutes before

**Comment by daschuer** [comment]
There is an issue with the path: 
```
Warning: Path Validation Error: Path(s) specified in the action for caching do(es) not exist, hence no cache is being saved.
```

**Comment by JoergAtGithub** [comment]
macOS x64 build time is 30 minutes again vs. 18 minutes in the last 2.4 build with PCH.

**Comment by daschuer** [comment]
That is expected, because we have no cache yet.

**Comment by JoergAtGithub** [comment]
Do you have any measurement values for macOS with and without PCH?

**Comment by daschuer** [comment]
> Do you have any measurement values for macOS with and without PCH?

No

**Comment by JoergAtGithub** [comment]
macOS caches seems to work now

**Comment by JoergAtGithub** [comment]
This is LGTM now, is the work on this PR  finished now?

**Comment by daschuer** [comment]
Yes, this is ready for merge.

**Comment by JoergAtGithub** [comment]
Thank you!

**Pull request merged**